deltacat/compute/compactor_v2/utils/primary_key_index.py

```diff
@@ -157,7 +157,12 @@ def _optimized_group_record_batches_by_hash_bucket(
 def group_by_pk_hash_bucket(
     table: pa.Table, num_buckets: int, primary_keys: List[str]
 ) -> np.ndarray:
-    table = generate_pk_hash_column([table], primary_keys, requires_sha1=True)[0]
+    new_tables = generate_pk_hash_column([table], primary_keys, requires_hash=True)
+    assert (
```

Comment:
What's the context of adding this new assertion?

Reply:
Something I felt should have been there to begin with.